code/game/machinery/computer/camera.dm

```diff
@@ -4,89 +4,140 @@
 	icon_screen = "cameras"
 	icon_keyboard = "security_key"
 	circuit = /obj/item/weapon/circuitboard/security
-	var/obj/machinery/camera/current = null
 	var/last_pic = 1
 	var/list/network = list("SS13")
 	var/mapping = 0//For the overview file, interesting bit of code.
+	var/list/watchers = list() //who's using the console, associated with the camera they're on.
 
 /obj/machinery/computer/security/check_eye(mob/user)
-	if ((get_dist(user, src) > 1 || user.eye_blind || !( current ) || !( current.status )) && (!istype(user, /mob/living/silicon)))
-		return null
-	var/list/viewing = viewers(src)
-	if((istype(user,/mob/living/silicon/robot)) && (!(viewing.Find(user))))
-		return null
-	user.reset_view(current)
-	return 1
+	if( (stat & (NOPOWER|BROKEN)) || user.incapacitated() || user.eye_blind )
+		user.unset_machine()
+		return
+	if(!(user in watchers))
+		user.unset_machine()
+		return
+	if(!watchers[user])
+		user.unset_machine()
+		return
+	var/obj/machinery/camera/C = watchers[user]
+	if(!C.can_use())
+		user.unset_machine()
+		return
+	if(!issilicon(user))
+		if(!Adjacent(user))
+			user.unset_machine()
+			return
+	else if(isrobot(user))
+		var/list/viewing = viewers(src)
+		if(!viewing.Find(user))
+			user.unset_machine()
 
+/obj/machinery/computer/security/on_unset_machine(mob/user)
+	watchers.Remove(user)
+	user.reset_perspective(null)

```

Comment:
I had to create this proc/on_unset_machine() to handle the special case that security camera console are. Having multiple people using it at once couldn't be handled with just check_eye() and unset_machine().